src/indexer/merger.rs

```diff
@@ -200,6 +200,12 @@ impl IndexMerger {
                 readers.push(reader);
             }
         }
+
+        if readers.is_empty() {
+            let err_msg = format!("Can't merge, all segments are empty {:?}.", segments);
+            return Err(crate::TantivyError::InvalidArgument(err_msg));
```

Comment:
Can we do something better?
Instead of returning an error, we want such a merge operation to succeed, not produce any new segment but remove the previous segments.

Reply:
I considered that, but `merge` seems to be the wrong operation that is triggered. Instead we could have another operation, like `remove` that is called when all segments are empty.
We should still safeguard the merge code, or allow it to be run on empty data.

Reply:
@PSeitz but  triggerring a merge over several segment and discoving that they all contain no docs is a legit case.

I don't think it needs to be a separate operation.

Reply:
I think that merge by definition creates a new segment, so should we create an empty segment?